### PR TITLE
doc(home-plugin): Adds documentation on <HomePageVisitedByType/> usage

### PR DESCRIPTION
**Pull request by aitherios**
## Hey, I just made a Pull Request!

This PR is part of a series with the objective of releasing a "Recently Visited" and "Top Visited" experience to open source. They are using the [home-page-visit-components-0](https://github.com/backstage/backstage/tree/home-page-visit-components-0) feature branch as an intermediary step before a PR to master.
The PRs in this series:
* [feature(home-plugin): Bootstrap Recently Visited and Top Visited components](https://github.com/backstage/backstage/pull/19645)
* [feature(home-plugin): Creates the VisitListener component](https://github.com/backstage/backstage/pull/19644)
* [feature(home-plugin): Create reference implementation (localStorage and storeApi)](https://github.com/backstage/backstage/pull/19643)
* [doc(home-plugin): Adds documentation on <HomePageVisitedByType/> usage](https://github.com/backstage/backstage/pull/19688)

A quick preview of the component in storybook:
Screenshot 2023-09-08 at 11 11 21

This patch is only about the plugin documentation on the component usage and requirements.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/home/README.md [hunk 1]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
 <CustomHomepageGrid config={defaultConfig}>
 ```
 
+## Page visit homepage component (HomePageVisitedByType)
+
+This component shows the homepage user a view for "Recently visited" or "Top visited".
+Being provided by the `<HomePageVisitedByType/>` component, see it in usage on a homepage example bellow:
```
Comment (adamdmharvey) [inline plugins/home/README.md:248]:
```suggestion
Being provided by the `<HomePageVisitedByType/>` component, see it in use on a homepage example below:
```

File: plugins/home/README.md [hunk 2]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
...
+      <HomePageVisitedByType kind="top" />
+    </Grid>
+    <Grid item xs={12} md={4}>
+      <HomePageVisitedByType kind="recent" />
+    </Grid>
+  </Grid>
+);
+```
+
+There ase some requirements to provide its functionality, so please ensure the following:
```
Comment (adamdmharvey) [inline plugins/home/README.md:267]: how did this PR pass spell check? :) something fishy

```suggestion
There are some requirements to provide its functionality, so please ensure the following:
```

File: plugins/home/README.md [hunk 3]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
...
+
+These components need an api to handle visit data, please refer to the [utility-apis](../../docs/api/utility-apis.md)
```
Comment (adamdmharvey) [inline plugins/home/README.md:269]:
```suggestion
These components need an API to handle visit data, please refer to the [utility-apis](../../docs/api/utility-apis.md)
```

**Review by adamdmharvey — COMMENTED** [review]
Few minor ones; this is a neat plugin!

File: plugins/home/README.md (hunk 2 above)
Comment (adamdmharvey) [reply plugins/home/README.md:267]: oh; since you aren't targeting master, i think that's why!

**Comment by aitherios** [comment]
Thanks for the help @adamdmharvey =D

File: plugins/home/README.md [hunk 4]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
 <CustomHomepageGrid config={defaultConfig}>
 ```
 
+## Page visit homepage component (HomePageVisitedByType)
+
+This component shows the homepage user a view for "Recently visited" or "Top visited".
```
Comment (camilaibs) [inline plugins/home/README.md:247]: Maybe clarify what a visit is. Is that a page visit?

File: plugins/home/README.md [hunk 5]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
...
+);
+```
+
+There are some requirements to provide its functionality, so please ensure the following:
+
+These components need an API to handle visit data, please refer to the [utility-apis](../../docs/api/utility-apis.md)
+documentation for more information. Bellow you can see an example for two options:
+
+```ts
+// apis.ts
```
Comment (camilaibs) [inline plugins/home/README.md:273]:
```suggestion
// packages/app/src/apis.ts
```

File: plugins/home/README.md [hunk 6]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
...
+    factory: ({ identityApi }) => LocalStoreVisitsApi.create({ identityApi }),
+  }),
+  // ...
+```
+
+To monitor page visit activity and save it on behalf of the user a component is provided, please add it to your app.
+See the example usage:
+
+```ts
+// App.tsx
```
Comment (camilaibs) [inline plugins/home/README.md:308]:
```suggestion
// packages/app/src/App.tsx
```

File: plugins/home/README.md [hunk 7]
```diff
@@ -242,6 +242,84 @@ const defaultConfig = [
 <CustomHomepageGrid config={defaultConfig}>
 ```
 
+## Page visit homepage component (HomePageVisitedByType)
+
+This component shows the homepage user a view for "Recently visited" or "Top visited".
+Being provided by the `<HomePageVisitedByType/>` component, see it in use on a homepage example below:
+
+```tsx
+import React from 'react';
```
Comment (camilaibs) [inline plugins/home/README.md:251]:
```suggestion
// packages/app/src/components/home/HomePage.tsx
import React from 'react';
```

**Review by camilaibs — APPROVED** [review]
LGTM 💯 , just left minor suggestions:

**Pull request merged**